pgmpy/factors/Factor.py

```diff
@@ -780,7 +780,9 @@ def __ne__(self, other):
         return not self.__eq__(other)
 
     def __hash__(self):
-        variables = sorted(self.variables)
+        var_with_hashes = list(map((lambda var: (hash(var), var)), self.variables))
```

Comment:
@ankurankan (hash(var), var) is required because we also need the variable ordering to generate the hash of Factor object.